libraries/AP_Networking/AP_Networking.cpp

```diff
@@ -0,0 +1,458 @@
+
+#include "AP_Networking.h"
+
+#if AP_NETWORKING_ENABLED
+
+#include "AP_Math/definitions.h"
+#include <GCS_MAVLink/GCS.h>
+
+#if CONFIG_HAL_BOARD == HAL_BOARD_CHIBIOS
+#include <hal_mii.h>
+#include <lwip/sockets.h>
+#else
+#include <arpa/inet.h>
+#include <sys/socket.h>
+
+#endif
+
+extern const AP_HAL::HAL& hal;
+
+const AP_Param::GroupInfo AP_Networking::var_info[] = {
+
+    // @Param: ENABLED
+    // @DisplayName: Enable Networking
```

Comment:
In general I think we should put "Networking" at the front of the Display name.  This will make the subsystem clearer for users.